src/ir/ty.rs

```diff
@@ -356,6 +357,39 @@ impl Type {
             _ => false,
         }
     }
+
+    /// If this type has a known size, return it (in bytes).
+    pub fn calc_size(&self, ctx: &BindgenContext) -> Option<usize> {
+        match self.kind {
+            TypeKind::Comp(ref ci) => {
+                ci.calc_layout(ctx).map(|layout| layout.size)
+            }
+            TypeKind::Enum(ref enum_ty) => {
+                enum_ty.calc_layout(ctx).map(|layout| layout.size)
+            }
+            TypeKind::Int(int_kind) => int_kind.known_size(),
+            TypeKind::Float(float_kind) => Some(float_kind.known_size()),
+            TypeKind::Complex(float_kind) => Some(float_kind.known_size() * 2),
+            TypeKind::Reference(..) |
+            TypeKind::NullPtr |
+            TypeKind::Pointer(..) |
+            TypeKind::BlockPointer => Some(mem::size_of::<*mut ()>()),
+            TypeKind::ResolvedTypeRef(inner) |
+            TypeKind::Alias(inner) |
+            TypeKind::TemplateAlias(inner, _) |
+            TypeKind::TemplateRef(inner, _) => {
+                ctx.resolve_type(inner).calc_size(ctx)
+            }
+            TypeKind::Array(inner, size) => {
+                ctx.resolve_type(inner).calc_size(ctx).map(|n| n * size)
+            }
+            TypeKind::Void |
+            TypeKind::Named |
+            TypeKind::Function(..) |
+            TypeKind::ObjCInterface(..) |
+            TypeKind::UnresolvedTypeRef(..) => None,
```

Comment:
I believe you can assume safely that all functions are pointer-sized, and use an `unreachable!` branch for `UnresolvedTypeRef`.